app/src/main/java/com/github/sdp_begreen/begreen/fragments/SendPostFragment.kt

```diff
@@ -143,10 +209,56 @@ class SendPostFragment : Fragment() {
                 // Display toast
                 Toast.makeText(requireContext(), R.string.photo_shared_success, Toast.LENGTH_SHORT)
                     .show()
+
+                // update the user's score in the active contests
+                it.trashCategory?.also { category ->
+                    it.location?.also { location ->
+                        contestsUpdateScores(category, location)
+                    }
+                }
             }
         }
     }
 
+    private suspend fun contestsUpdateScores(
```

Comment:
This method is quite long and does several things. Maybe you could refactor in two methods like that ? : 

private suspend fun contestsUpdateScores(trashCategory: TrashCategory, location: CustomLatLng) {
    val userId = connectedUserViewModel.currentUser.value?.id
    val participationMap = eventsFragmentViewModel.participationMap.dropWhile { it.isEmpty() }.first()
    val contests = eventsFragmentViewModel.allEvents.dropWhile { it.isEmpty() }.first()

    contests.forEach { contest ->
        processContest(contest, userId, participationMap, trashCategory, location)
    }
}

private suspend fun processContest(
    contest: Contest,
    userId: String?,
    participationMap: Map<String, Boolean>,
    trashCategory: TrashCategory,
    location: CustomLatLng
) {
    location.toMapLocation()?.let {
        if (participationMap[contest.id] == true && contest.isActive() == true && contest.isInRange(it)) {
            val updatedParticipant = eventParticipantService.getParticipant(
                RootPath.CONTESTS,
                contest.id!!,
                userId!!,
                ContestParticipant::class.java
            )
            eventParticipantService.addParticipant(
                RootPath.CONTESTS,
                contest.id!!,
                updatedParticipant.copy(
                    score = updatedParticipant.score?.plus(trashCategory.value)
                        ?: trashCategory.value
                )
            )
        }
    }
}